src/service/url-replacements-impl.js

```diff
@@ -372,24 +406,10 @@ export class GlobalVariableSource extends VariableSource {
     );
 
     // Returns assigned geo value for geoType or all groups.
-    this.setAsync(
+    this.setBoth(
       'AMP_GEO',
-      /** @type {AsyncResolverDef} */ (
-        (geoType) => {
-          return this.getGeo_((geos) => {
-            if (geoType) {
-              userAssert(
-                geoType === 'ISOCountry',
-                'The value passed to AMP_GEO() is not valid name:' + geoType
-              );
-              return /** @type {string} */ (geos[geoType] || 'unknown');
-            }
-            return /** @type {string} */ (
-              geos.matchedISOCountryGroups.join(GEO_DELIM)
-            );
-          }, 'AMP_GEO');
-        }
-      )
+      (geoType) => geoData(this.cachedGeo_, geoType),
```

Comment:
I'm not too familiar with `url-replacements`, but isn't this a race condition? Is it fine to default to `unknown` before amp-geo service finishes loading?

Reply:
Yes, it's a race condition, but intentionally.

We opt to not interfering with delays when a user clicks on a link targetted by replacement by waiting for async data. If the data is not present for sync access here, a default fallback value is used.

Likely, if we do this, it will need some additional documentation.

Reply:
Apologies if this is a silly question, but why does it need to be sync? Why can't the URL resolve async?

Reply:
If the value is async, then users have to wait during a click event to start a navigation to the next document.

(I think we could also rewrite async items greedily as an alternative, but this isn't supported right now)